pkg/cli/config/config.go

```diff
@@ -96,6 +98,52 @@ func GetBackplaneConfiguration() (bpConfig BackplaneConfiguration, err error) {
 	return bpConfig, nil
 }
 
+func getProxyURL(bpConfig BackplaneConfiguration, proxyURLConfig any) string {
+	// as part of OSD-19785, proxy-url in config file is converted to []string from string to allow multiple proxies.
+	// For backward compatibility, get proxy-url as 'any' and check if it is still string (set in past) or array
+	// if string, return it else if array, return the first working proxy
+	switch t := proxyURLConfig.(type) {
+	case []interface{}:
+		logger.Infof("proxy-url array found in config file: %s ", t)
+		return getFirstWorkingProxyURL(bpConfig, proxyURLConfig.([]interface{}))
+	case interface{}:
+		logger.Warn("Change proxy-url in config file to an array. Multiple proxy-urls are now supported. First working proxy-url from the array will be used")
+		return proxyURLConfig.(string)
+	default:
+		logger.Warnf("Unrecognizable value type for proxy-url in config file: %s", t)
+		return ""
+	}
+}
+
+func getFirstWorkingProxyURL(bpConfig BackplaneConfiguration, s []interface{}) string {
```

Comment:
may be 
```
 func (config *BackplaneConfiguration) checkProxyHealth(string proxyURL) bool {
    ... check the proxy health for both cases ( array or string).....
 }
```

Reply:
The method needs to return first working proxy from the configured multiple proxies. I have changed code such that iif only one proxy is configured, it is converted to slice and passed to this method. Please elaborate if you meant something else.